apps/wallet/src/ui/app/hooks/useQredoTransaction.tsx

```diff
@@ -0,0 +1,83 @@
+import { Heading } from '../shared/heading';
+import { Text } from '../shared/text';
+
+export function useQredoTransaction() {
+    const [clientIdentifier, setClientIdentifier] = useState(() => uuidV4());
+    const [qredoTransactionID, setQredoTransactionID] = useState<string | null>(
+        null
+    );
+    const notificationModal = (
+        <ModalDialog
+            isOpen={!!qredoTransactionID}
+            body={
+                <div className="flex flex-col gap-2.5 text-center items-center relative">
+                    <div className="bg-[url('_assets/images/qredo.png')] h-14 w-14 bg-cover" />
+                    <div className="text-steel">
+                        <LoadingIndicator color="inherit" />
+                    </div>
+                    <Heading variant="heading6" color="gray-90" weight="medium">
+                        Awaiting transaction approval in the Qredo app
+                    </Heading>
+                    <Text variant="pBodySmall" color="steel-dark">
+                        Check your Qredo app for status. Once all required
+                        custody approvals have been performed the transaction
+                        will complete.
+                    </Text>
+                    <Button
+                        text="Close"
+                        onClick={() => {
+                            QredoEvents.emit('clientIgnoredUpdates', {
+                                clientIdentifier,
+                            });
+                            setQredoTransactionID(null);
+                        }}
+                    />
+                </div>
+            }
+            onClose={() => {
+                QredoEvents.emit('clientIgnoredUpdates', {
+                    clientIdentifier,
+                });
+                setQredoTransactionID(null);
+            }}
+        />
+    );
+    useEffect(() => {
+        function qredoTransactionCreated(
+            event: QredoEventsType['qredoTransactionCreated']
+        ) {
+            if (event.clientIdentifier === clientIdentifier) {
+                setQredoTransactionID(event.qredoTransaction.txID);
+            }
+        }
+        function qredoTransactionComplete(
+            event: QredoEventsType['qredoActionDone']
+        ) {
+            if (event.clientIdentifier === clientIdentifier) {
+                setQredoTransactionID(null);
+                setClientIdentifier(uuidV4());
+            }
+        }
+        QredoEvents.on('qredoTransactionCreated', qredoTransactionCreated);
+        QredoEvents.on('qredoActionDone', qredoTransactionComplete);
+        return () => {
+            QredoEvents.off('qredoTransactionCreated', qredoTransactionCreated);
+            QredoEvents.off('qredoActionDone', qredoTransactionComplete);
+            QredoEvents.emit('clientIgnoredUpdates', { clientIdentifier });
+        };
+    }, [clientIdentifier]);
+    return {
+        clientIdentifier,
```

Comment:
What if you send two qredo transactions during a session, could that possible result int he same client identifier?